### PR TITLE
feat: timestamp utility — wall-clock capture and display

### LINKED ISSUE #76: Timestamp utility — capture and display local wall-clock time
## Summary
Utility functions for capturing local wall-clock time when an entry is saved, and formatting timestamps for display. Ensures entries always reflect the time they occurred in the user's day, regardless of future timezone changes (user story #69).

## Acceptance criteria
- [ ] `nowLocalIso()` — returns current time as an ISO 8601 string with UTC offset (e.g. `2026-03-04T09:00:00-08:00`); this is what gets stored in `entry.timestamp`
- [ ] `formatEntryTime(isoString)` — returns a display string (e.g. `9:00 AM`) parsed from the stored offset, not re-interpreted in the current timezone
- [ ] `formatEntryDate(isoString)` — returns a date header string (`Today`, `Yesterday`, `March 2`, etc.) relative to the device's current local date, but using the stored date component from the ISO string
- [ ] If the user travels to a new timezone, previously stored entries display at the same wall-clock time they were logged
- [ ] Tests: unit tests covering cross-timezone display stability (store in PST, display in CET → same time shown)

## Notes
Affected files: new `src/lib/utils/timestamp.ts`
All entry screens use `nowLocalIso()` on submit. Trace uses `formatEntryTime` and `formatEntryDate` for display.
Contributes to #69
Blocked by: nothing

### PR DESCRIPTION
**Pull request by sfrankle**
## Summary

- Adds `nowLocalIso()` to capture the current wall-clock time plus UTC offset as an ISO 8601 string (`YYYY-MM-DDTHH:mm:ss±HH:MM`). This avoids the `toISOString()` / UTC-Z pitfall where an entry logged at 9:00 AM in one timezone displays as a different time after the user or device moves.
- Adds `formatEntryTime(isoString)` to parse and display the stored wall-clock hours/minutes as 12-hour AM/PM, using the stored offset — not the device's current timezone.
- Adds `formatEntryDate(isoString, _today?)` to render "Today", "Yesterday", or "Month D" using the stored date component, also immune to timezone re-interpretation.

Closes #76
Contributes to #69

## What to review

- `src/lib/utils/timestamp.ts` — three named exports, no external dependencies, pure JS Date + string manipulation + Intl.DateTimeFormat
- `src/__tests__/utils/timestamp.test.ts` — 16 unit tests written before implementation (TDD); key cases are the cross-timezone stability tests for both time and date display, and the `_today` injection parameter on `formatEntryDate`

## Testing

- 16 Jest unit tests; all pass
- Tests cover: ISO format regex, offset correctness, 12-hour AM/PM edge cases (midnight, noon), half-hour UTC offsets, cross-timezone wall-clock stability, Today/Yesterday/Month-day labels, injected today for deterministic date comparison
- No Maestro flow tests — this is a utility with no UI; flow tests belong to the entry screen tasks that consume it
- No schema changes

## Checklist
- [x] Acceptance criteria met
- [x] Flow tests (Maestro) added or updated for any user-facing behavior — N/A (utility only)
- [x] Migration test written if schema changed — N/A (no schema change)
- [ ] CI passing
- [x] `docs/changelog.md` updated
- [x] No judgmental language in UI strings
- [x] No network calls or off-device data transmission introduced

### CODE REVIEW

File: src/lib/utils/timestamp.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+/**
+ * Returns the current local wall-clock time as an ISO 8601 string with the
+ * device's current UTC offset, e.g. "2026-03-09T09:00:00-08:00".
+ *
+ * Does NOT use toISOString() (which returns UTC with Z suffix) — that would
+ * lose the wall-clock time when displayed after timezone changes.
+ */
+export function nowLocalIso(): string {
+  const d = new Date();
```
[the remark this answers is not in the document]
Comment (sfrankle) [reply src/lib/utils/timestamp.ts]: Totally fair — it looks like it should be simpler. The short version: JS has no built-in that returns 'current local time with offset attached.' `new Date().toISOString()` returns UTC (`2026-03-04T17:00:00Z`), which is the exact bug we're avoiding — an entry logged at 9 AM would display as 5 PM after a timezone change.

So we have to manually assemble the ISO string from local component methods (`getHours`, `getMinutes`, etc.) plus `getTimezoneOffset()`. That's what the 8 lines do — it's verbose but there's no shortcut in the JS standard library without pulling in a date library like luxon or dayjs.

If you'd prefer to add a dependency to hide this complexity, that's a valid call. Otherwise this is the minimum needed to get the behavior right.
